q2_moshpit/eggnog/tests/test_utils.py

```diff
@@ -0,0 +1,134 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+from unittest.mock import patch, call, MagicMock
+from qiime2.plugin.testing import TestPluginBase
+from qiime2.core.exceptions import ValidationError
+from .._utils import (
+    _download_and_build_hmm_db, _download_fastas_into_hmmer_db,
+    _merge_hmms_and_write_idmap, COMMON_URL, _validate_eggnog_hmmer_taxon_id
+)
+from q2_moshpit.eggnog._format import EggnogHmmerIdmapDirectoryFmt
+from q2_types.genome_data import ProteinsDirectoryFormat
+from q2_types.profile_hmms import (
+    PressedProfileHmmsDirectoryFmt, ProteinMultipleProfileHmmDirectoryFmt
+)
+
+
+class TestEggnogUtils(TestPluginBase):
+    package = "q2_moshpit.eggnog.tests"
+
+    @patch("urllib.request.urlopen")
+    def test_validate_eggnog_hmmer_taxon_id_valid(self, mock_urlopen):
+        # Mock HTML content
+        mock_html = """
+        <html>
+            <body>
+                <a href="1/">1/</a>
+                <a href="2/">2/</a>
+                <a href="3/">3/</a>
+            </body>
+        </html>
+        """
+        mock_response = MagicMock()
+        mock_response.read.return_value = mock_html.encode('utf-8')
+        mock_urlopen.return_value.__enter__.return_value = mock_response
+
+        _validate_eggnog_hmmer_taxon_id(2)
+
+    @patch("urllib.request.urlopen")
+    def test_validate_eggnog_hmmer_taxon_id_invalid(self, mock_urlopen):
+        # Mock HTML content
+        mock_html = """
+        <html>
+            <body>
+                <a href="1/">1/</a>
+                <a href="2/">2/</a>
+                <a href="3/">3/</a>
+            </body>
+        </html>
+        """
+        mock_response = MagicMock()
+        mock_response.read.return_value = mock_html.encode('utf-8')
+        mock_urlopen.return_value.__enter__.return_value = mock_response
+
+        with self.assertRaisesRegex(
+            ValidationError, "4 is not a valid taxon ID."
+        ):
+            _validate_eggnog_hmmer_taxon_id(4)
+
+    @patch("subprocess.run")
+    @patch("tempfile.TemporaryDirectory")
+    def test_download_and_build_hmm_db(self, tmpdir, mock_run):
```

Comment:
I made a comment right below about adding more content assertions and this applies here too - maybe we can at least `.validate()`?